engine/title.asm

```diff
@@ -3,17 +3,654 @@ INCLUDE "constants.asm"
+	call IncreaseSequenceNumber ;6025
+	ld a, $01
+	ldh [$FFDE], a
+	ret	
+	
+Start:: ;6045
+	call IncreaseSequenceNumber
+	push de
+	ld de, $002D
+	call PlaySFX	; Play "Swish" sound
+	pop de
+	ld a, $80
+	ld [wSequence+2], a
+	call SetLYOverrides
+	ld a, $43
+	ldh [$FFD0], a
+	ret
+
+MoveTitle:: ;605d
+	xor a
+	ldh [$FFDE], a
+	ld hl, wSequence+2
+	ld a, [hl]
+	and a
+	jr z, .movetitle_nextseq
+	add $04
+	ld [hl], a
+	ld e, a
+.movetitle_wait
+	ldh a, [rLY]
+	cp $40
+	jr c, .movetitle_wait
+	ld a, e
+	call SetLYOverrides
+	ret
+
+.movetitle_nextseq
+	call IncreaseSequenceNumber
+	ret	
+	
+MoveTitleEnd:: ;607a
+	xor a
+	ldh [$FFD0], a
+	call IncreaseSequenceNumber
+	ld de, $0001
+	call PlayMusic		; Play Title Theme
+	ret
+
+InitFlashTitle:: ;6087
+	call IncreaseSequenceNumber
+	ld a, $1A
+	ld [wSequence+2], a
+	ld a, $06
+	ld [wSequence+3], a
+	ret
+
+FlashTitle:: ;6095
+	ld hl, wSequence+3
+	ld a, [hl]
+	and a
+	jr z, .exit
+	dec [hl]
+	ld a, [wSequence+2]
+	xor $1A
+	ld [wSequence+2], a
+	ldh [rBGP], a
+	call DelayFrame
+	call DelayFrame
+	ret
+
+.exit
+	call IncreaseSequenceNumber
+	ld a, $E4
+	ldh [rBGP], a
+	ret
+
+PMJapaneseChara:: ;60b6
+	call PrintPMJapaneseChara
+	ld a, $10
+	ld [wSequence+2], a
+	call IncreaseSequenceNumber
+	ld a, $01
+	ldh [$FFDE], a
+	ret
+
+PMSubtitle:: ;60c6
+	call PrintPMSubtitle
+	ld a, $10
+	ld [wSequence+2], a
+	call IncreaseSequenceNumber
+	ld a, $01
+	ldh [$FFDE], a
+	ret
+
+Version:: ;60d6
+	call PrintVersion
+	ld a, $10
+	ld [wSequence+2], a
+	call IncreaseSequenceNumber
+	ld a, $01
+	ldh [$FFDE], a
+	ret
+
+CopyRight:: ;60e6
+	call PrintCopyRight
+	ld a, $10
+	ld [wSequence+2], a
+	call IncreaseSequenceNumber
+	ld a, $01
+	ldh [$FFDE], a
+	ret
+
+HoOh:: ;60f6
+	call SetHoOh
+	ld a, $10
+	ld [wSequence+2], a
+	call IncreaseSequenceNumber
+	ld a, $01
+	ldh [$FFDE], a
+	ret
+
+PressButtonInit:: ;6106
+	ld hl, wSequence
+	inc [hl]
+	ld hl, wSequence+2
+	ld de, DecodeNybble0Table - 3	; DecodeNybble0Table - 3 = $0C00
+	ld [hl], e
+	inc hl
+	ld [hl], d
+	ret	
+	
+TitleScreenInputAndTimeout:: ;6114
+	ld hl, wSequence+2
+	ld e, [hl]
+	inc hl
+	ld d, [hl]
+	ld a, e
+	or d
+	jr z, .psbtn_reset
+	dec de
+	ld [hl], d
+	dec hl
+	ld [hl], e
+	call GetJoypad
+	ld hl, hJoyState	; hJoyState = $FFA3
+	ld a, [hl]
+	and $46				; UP + B + SELECT brings you to the SRAM clear screen.
+	cp $46
+	jr z, .psbtn_sramclear
+	ld a, [hl]
+	and $04				; SELECT will bring you to the debug menu.
+	jr nz, .psbtn_gotodebug
+	ld a, [hl]
+	and $09
+	ret z
+	
+.psbtn_play
+	ld a, $00			;MainMenu
+	jr .psbtn_nextseq
+
+.psbtn_gotodebug
+if DEBUG 
+	ld a, $01			;DebugMenu
+	jr .psbtn_nextseq
+else
+	ret
+endc
+
+.psbtn_sramclear
+	ld a, $02			;SRAMClearMenu
+	
+.psbtn_nextseq
+	ld [wSequence + 1], a	; wSequence + 1 = $CB5F
+	ld hl, wSequence	; wSequence = $CB5E
+	set 7, [hl]
+	ret
+
+.psbtn_reset
+	ld hl, wSequence	; wSequence = $CB5E
+	inc [hl]
+	xor a
+	ld [wNextBGM], a ;c1a7
+	ld [wNextBGM + 1], a ;c1a8
+	ld hl, wSoundFade ;c1a5
```

Comment:
`wNextBGM` is `wMusicFade` in [pokecrystal/wram.asm](https://github.com/pret/pokecrystal/blob/7307fc8dce605823fbdf8e5ec8530d7aaf675cfa/wram.asm#L82). However, `wSoundFade` cannot be extrapolated to the given label in pokecrystal, which is `wSFXNoiseSampleSet`.

Reply:
This is definitely wNextBGM/wSoundFade. wNextBGM changes when you go into different fields to the proper music id. wSoundFade and the byte immediately after it are related (SoundFade seems to be the sound fade "mode", c1a6 increases/decreases to 8 when the track fades in/out)